src/convolution.cpp

```diff
@@ -739,6 +753,32 @@ std::size_t ConvolutionDescriptor::ForwardBackwardDataGetWorkSpaceSizeDirect(
     }
 }
 
+std::size_t ConvolutionDescriptor::ForwardBackwardDataGetWorkSpaceSizeWinograd(
+    const miopen::ConvolutionContext& ctx) const
+{
+    if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_WINOGRAD{}))
+        return 0;
+
+    try
```

Comment:
Why should something like FindAllSolutions or GetWorkspaceSize throw any exception?

Reply:
For uniformity of all GetWorkspaceSizes. 
In the future all functions must be merged into single one

Reply:
Still, why should GetWorkspaceSize ever throw?

Reply:
Some of them may throw. Some GEMMs have definitely done this before
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/6549d8e36aff494317cf3d4bc19597fb094c54d2/src/reducetensor.cpp#L286-L294

Reply:
> Please add an issue about removing exceptions here. 

Yep, it looks like we do not trust in our own code ;) Perhaps we shall keep try/catch for Release builds, but allow the library to fail in Debug builds. This is common pattern that worth dedicated ticket.

> If some checks like inDescLengths.size() != outDescLengths.size() are needed it should be done in appropriate place and GetWorkspaceSize() doesn't look like appropriate place. 

I am second to this. Strange check, really ;)

Reply:
@Kirpich30000 After second thought:

> If some checks like inDescLengths.size() != outDescLengths.size() are needed it should be done in appropriate place and GetWorkspaceSize() doesn't look like appropriate place.

I am second to this but this is not relevant to this PR. `ForwardBackwardDataGetWorkSpaceSizeWinograd()` does not have inappropriate checks.

> Please add an issue about removing exceptions here.

When a solver is unable to solve a problem, it's `IsApplicable()` must return `false`. And `IsApplicable()` must not throw. Exception there means programming error that needs to be fixed. And try/catch here looks like we are hiding these errors.

However this is not exactly so. First of all, it is theoretically impossible to fully test the library. This means that we can deliver a library that contains elusive programming errors, including exceptions in `IsApplicable()`. The `try/catch` blocks in GWSS functions (a common pattern in the library) are intended to improve stability of the library in the event of such errors.

Second, we are not hiding errors, but lowering the level. Instead of termination, the library logs an Error onto console for Debug builds, and Warning for Releases.

If you insist, the only thing I have in mind is removing `try/catch` for Debug builds.